mqtt/mqtt-broker/src/broker.rs

```diff
@@ -103,7 +103,27 @@ where
         BrokerState { retained, sessions }
     }
 
-    fn process_message(&mut self, client_id: ClientId, event: ClientEvent) -> Result<(), Error> {
+    pub fn clone_state(&self) -> BrokerState {
```

Comment:
do we want this in release bits? can we guard this with `#[cfg(any(test, feature = "proptest"))]`

Reply:
Something is telling me that we may want to use this method in integration tests maybe. Not 100% sure though